// swagger-autogen-doc/parser/grammar/swagger-gen/SwaggerParserListener.js
// Generated from /home/alex/Project/NodeJs/swagger-autogen-doc/swagger-autogen/src/parser/grammar/SwaggerParser.g4 by ANTLR 4.9.1
// jshint ignore: start
const antlr4 = require('antlr4');

// This class defines a complete listener for a parse tree produced by SwaggerParser.
module.exports = class SwaggerParserListener extends antlr4.tree.ParseTreeListener {

	// Enter a parse tree produced by SwaggerParser#swaggerDocument.
	enterSwaggerDocument(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerDocument.
	exitSwaggerDocument(ctx) {
	}


	// Enter a parse tree produced by SwaggerParser#swaggerElements.
	enterSwaggerElements(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerElements.
	exitSwaggerElements(ctx) {
	}


	// Enter a parse tree produced by SwaggerParser#swaggerElement.
	enterSwaggerElement(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerElement.
	exitSwaggerElement(ctx) {
	}


	// Enter a parse tree produced by SwaggerParser#swaggerContent.
	enterSwaggerContent(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerContent.
	exitSwaggerContent(ctx) {
	}


	// Enter a parse tree produced by SwaggerParser#swaggerAttribute.
	enterSwaggerAttribute(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerAttribute.
	exitSwaggerAttribute(ctx) {
	}


	// Enter a parse tree produced by SwaggerParser#swaggerChardata.
	enterSwaggerChardata(ctx) {
	}

	// Exit a parse tree produced by SwaggerParser#swaggerChardata.
	exitSwaggerChardata(ctx) {
	}



}
