// swagger-autogen-doc/parser/grammar/swagger-gen/SwaggerParserVisitor.js
// Generated from /home/alex/Project/NodeJs/swagger-autogen-doc/swagger-autogen/src/parser/grammar/SwaggerParser.g4 by ANTLR 4.9.1
// jshint ignore: start
const antlr4 = require('antlr4');

// This class defines a complete generic visitor for a parse tree produced by SwaggerParser.

module.exports = class SwaggerParserVisitor extends antlr4.tree.ParseTreeVisitor {

	// Visit a parse tree produced by SwaggerParser#swaggerDocument.
	visitSwaggerDocument(ctx) {
	  return this.visitChildren(ctx);
	}


	// Visit a parse tree produced by SwaggerParser#swaggerElements.
	visitSwaggerElements(ctx) {
	  return this.visitChildren(ctx);
	}


	// Visit a parse tree produced by SwaggerParser#swaggerElement.
	visitSwaggerElement(ctx) {
	  return this.visitChildren(ctx);
	}


	// Visit a parse tree produced by SwaggerParser#swaggerContent.
	visitSwaggerContent(ctx) {
	  return this.visitChildren(ctx);
	}


	// Visit a parse tree produced by SwaggerParser#swaggerAttribute.
	visitSwaggerAttribute(ctx) {
	  return this.visitChildren(ctx);
	}


	// Visit a parse tree produced by SwaggerParser#swaggerChardata.
	visitSwaggerChardata(ctx) {
	  return this.visitChildren(ctx);
	}



}
